src/coprocessor/dag/executor/index_scan.rs

```diff
@@ -176,11 +180,18 @@ impl<S: Store> IndexScanExecutor<S> {
 
 impl<S: Store> Executor for IndexScanExecutor<S> {
     fn next(&mut self) -> Result<Option<Row>> {
+        let now = Instant::now();
         loop {
             if let Some(row) = self.get_row_from_range_scanner()? {
                 if let Some(counts) = self.counts.as_mut() {
                     counts.last_mut().map_or((), |val| *val += 1);
                 }
+                self.run_time_stats.record_and_set(
```

Comment:
I prefer to find some ways (i.e. some kind of scope guard) to avoid writing these statements twice because we may add new exit point and we may forget to add these lines at every exit code in future.

Reply:
There may be a problem here. Using scope guard will need to mutable borrow `self` since we need to modify `run_time_stats`, but the remaining codes also need to modify self and need mutable borrow as well.

Reply:
You can have a try. There can be multiple places using a &mut self as long as the function call does not generate a new longer lifetime reference